### PR TITLE
Remove dependency on google fonts.

### PR DESCRIPTION
**Pull request by dbalan**
**warning** not a front-end dev, just slightly annoyed

TL;DR serve fonts from hillhacks.in server.

The fonts are imported from google fonts archive[1], and this change
removes client call to google servers.

The fonts themselves have permissive licenses afiu, of which you can
find in the respective folders.

In future, if one wants to import a google font, get it from [1] and
add the @font-face bits to _sass/_fonts.scss

[1] https://github.com/google/fonts

### CODE REVIEW

**Comment by ppaeps** [comment]
On 2017-03-29 07:53:18 (-0700), Dhananjay Balan <notifications@github.com> wrote:
> **warning** not a front-end dev, just slightly annoyed
> TL;DR serve fonts from hillhacks.in server.

This seems like a good idea but I worry about the additional latency for
people far away from where hillhacks.in is hosted.

The font files are quite large and the latency is already unpleasant.

This is obviously the right thing to do, but will it make any difference
at all?

Philip

-- 
Philip Paeps
Senior Reality Engineer
Ministry of Information

**Comment by bsva** [comment]
On Wednesday 29 March 2017 05:20 PM, Philip Paeps wrote:
> On 2017-03-29 07:53:18 (-0700), Dhananjay Balan <notifications@github.com> wrote:
>> TL;DR serve fonts from hillhacks.in server.
> This is obviously the right thing to do, but will it make any difference
> at all?

Are you saying that it doesnt matter if google gets to know about any
hillhacks.in website visitor?

Who are you, and what have you done to Trouble? ;)

Cheers,
sva.

**Comment by captn3m0** [comment]
An alternate solution might be to remove the dependency on external fonts and rely on the [system font stack](https://css-tricks.com/snippets/css/system-font-stack/) instead.

But I really like the Francois One though. We can drop the remaining, and just load that one from our servers?

**Comment by ppaeps** [comment]
On 2017-03-29 17:29:14 (+0200), Nemo <notifications@github.com> wrote:
> An alternate solution might be to remove the dependency on external
> fonts and rely on the [system font 
> stack](https://css-tricks.com/snippets/css/system-font-stack/) 
> instead.

That sounds like a good idea.

> But I really like the Francois One though. We can drop the remaining,
> and just load that one from our servers?

Sure.

I don't mind either way.  I just want to keep the latency manageable.

Philip

-- 
Philip Paeps
Senior Reality Engineer
Ministry of Information

**Comment by dbalan** [comment]
```
find fonts -type f -name '*.ttf' -exec cat {} \; | gzip -c -f | wc -c
783371
```
 
Its around 800k, I agree with you that its a bit hard for people on
slow connections . google does not really set any cookies as far as I
can see but http:// HEAD would be enough them to build a user profile
(as far as I can see the font-urls are generated random).

I like the idea of not depending on too much external fonts, another naive approach that I've is to actually host in India - considering the distribution of users.

**Comment by ppaeps** [comment]
On 2017-03-29 08:40:14 (-0700), Dhananjay Balan <notifications@github.com> wrote:
> ```
> find fonts -type f -name '*.ttf' -exec cat {} \; | gzip -c -f | wc -c
> 783371
> ```
>  
> Its around 800k, I agree with you that its a bit hard for people on
> slow connections.

It's also a sizeable percentage of the total data.  Actual content pales
in comparison.

> google does not really set any cookies as far as I can see but http://
> HEAD would be enough them to build a user profile (as far as I can see
> the font-urls are generated random).
> 
> I like the idea of not depending on too much external fonts, another
> naive approach that I've is to actually host in India - considering
> the distribution of users.

I think Nemo's suggestion to use default fonts and serve only the pretty
one locally is a lot more likely to get implemented quickly.  Moving
machines is hard work and it's complicated enough as it is.

Philip

-- 
Philip Paeps
Senior Reality Engineer
Ministry of Information

**Comment by captn3m0** [comment]
The sans-serif fonts in content works out fine, including headlines:

(Robot for me, which I have locally installed)

image

The nav looks fine:

image

This is "hopefully okay":
in the default monospace font (source code pro) for me

image

I checked in courier and it was fine as well. Committed to this branch, so we can merge if this is good with everyone?

**Comment by ppaeps** [comment]
Yeah... Looks good to me. 

Keep it simple.

Philip

-- 
Philip Paeps
Senior Reality Engineer
Ministry of Information

**Comment by dbalan** [comment]
@captn3m0 Thanks for fixing! You are a lightsaber!

**Comment by captn3m0** [comment]
Did some more yak-shaving: https://github.com/hillhacks/website/pull/5
